src/routes/transactions/tests/preview.rs

```diff
@@ -0,0 +1,141 @@
+use std::time::Duration;
+
+use mockall::predicate::eq;
+use rocket::http::{ContentType, Header, Status};
+use rocket::local::asynchronous::Client;
+use serde_json::json;
+
+use crate::common::models::backend::chains::{
+    BlockExplorerUriTemplate, ChainInfo, GasPrice, NativeCurrency, RpcAuthentication, RpcUri, Theme,
+};
+use crate::config::contract_info_request_timeout;
+use crate::providers::info::*;
+use crate::routes::contracts::models::DataDecoderRequest;
+use crate::routes::transactions::handlers::preview::TransactionPreview;
+use crate::routes::transactions::tests::PREVIEW_RESPONSE;
+use crate::tests::main::setup_rocket;
+use crate::utils::errors::ApiResult;
+use crate::utils::http_client::{MockHttpClient, Request, Response};
+
+fn get_chain_info() -> ChainInfo {
```

Comment:
Wouldn't it be more consistent to have this as a json file, like we do with the transactions?

Reply:
This should work but I'd need to move the "mock" to the httpclient (which I also prefer)

Reply:
The mock on the InfoProvider was done on many other tests. I'd need to review why it was done like that